### PR TITLE
HBW-164 Add table for claiming

### PR DESCRIPTION
**Pull request by mxsmirnoff**

### CODE REVIEW

**Comment by mxsmirnoff** [comment]
image
image

File: hbw/app/javascript/packs/hbw/components/claiming/table.js.jsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+              <th>{translator('claiming_table.title')}</th>
+              <th>{translator('claiming_table.type')} </th>
+              <th>{translator('claiming_table.description')}</th>
+              <th>{translator('claiming_table.sla')}</th>
+              {unassigned && <th></th>}
+            </tr>
+            </thead>
+            <tbody ref={(t) => { this.tableBody = t; }}>
+            {tasks
+              .sort((a, b) => b.priority - a.priority)
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/claiming/table.js.jsx]: We have agreed to sort all stuff in the backend.

File: hbw/app/javascript/packs/hbw/components/claiming/table.js.jsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+              .map((row, index) => this.renderRow(row, index))}
+            </tbody>
+          </table>
+          {fetching && <div className='loader'><i className="fas fa-spinner fa-spin fa-2x"></i></div>}
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/claiming/table.js.jsx]: Move to `renderLoader`

File: hbw/app/javascript/packs/hbw/components/claiming/table.js.jsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+          {!fetching && tasks.length === 0
+            && <div className='no-tasks'><span>{translator('claiming_table.empty_tasks')}</span></div>}
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/claiming/table.js.jsx]: Move to `renderNoTasks`

File: hbw/app/javascript/packs/hbw/components/claiming/priority.js.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+modulejs.define('HBWPriority', ['React'], (React) => {
```
[the remark this answers is not in the document]
Comment (sugarfree1) [reply hbw/app/javascript/packs/hbw/components/claiming/priority.js.jsx]: sorry, missed it

**Comment by mxsmirnoff** [comment]
image
With @sugarfree1 decided to do `Urgent` red for now and `High` yellow. 
Question about colours is written on the board.

**Comment by mxsmirnoff** [comment]
Updated

File: hbw/app/javascript/packs/hbw/components/helpers/unassigned_tasks.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  createUnassignedSubscription = (firstResult, maxResults) => {
+    const data = {
+      entity_class: this.props.env.entity_class,
+      first_result: firstResult,
+      max_results:  maxResults,
+      unassigned:   true
+    };
+
+
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/helpers/unassigned_tasks.js]: extra line

File: hbw/app/javascript/packs/hbw/components/helpers/unassigned_tasks.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+
+export default WrappedComponent => class WithUnssignedTasks extends Component {
+  generateGuid = () => `hbw-${Math.floor(Math.random() * 0xFFFF)}`;
+
+  state = {
+    guid:         this.generateGuid(),
+    pollInterval: this.props.env.poll_interval,
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/helpers/unassigned_tasks.js]: Do you really want to change poll interval for existing subscription?

Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/helpers/unassigned_tasks.js]: BTW propTypes?

File: hbw/app/javascript/packs/hbw/components/helpers/unassigned_tasks.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      data
+    })
+      .syncing(() => this.setState({ syncing: true }))
+      .progress(() => this.setState({ error: null }))
+      .fail(response => this.setState({ error: response }))
+      .always(() => this.setState({ syncing: false }));
+  };
+
+  render () {
+    return <WrappedComponent generateGuid={this.generateGuid}
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/helpers/unassigned_tasks.js]: Why do you want to provide this function to wrapped component?

**Comment by mxsmirnoff** [comment]
Updated

**Comment by mxsmirnoff** [comment]
Updated

**Pull request merged**